subsys/bluetooth/audio/unicast_client.c

```diff
@@ -1943,10 +2147,10 @@ static uint8_t unicast_client_pacs_context_read_func(struct bt_conn *conn,
 		BT_DBG("pac %p context 0x%04x", pac, pac->context);
 	}
 
-discover_ase:
+discover_loc:
 	/* Read ASE instances */
-	if (unicast_client_ase_discover(conn, params) < 0) {
-		BT_ERR("Unable to read ASE");
+	if (unicast_client_pacs_location_discover(conn, params) < 0) {
```

Comment:
Are you going to make use of the locations somewhere in the stack?
Maybe it's not needed to discover and read locations if user did not registered `unicast_client_cbs->location` callback?

Reply:
> Maybe it's not needed to discover and read locations if user did not registered `unicast_client_cbs->location` callback?

Hmm, yeah I guess that if the location callback wasn't set, we could skip it (that is seemingly OK with the spec). In mos cases the callback structure will be constant and not changing, but there's technically nothing stopping the application from registering the callbacks with `unicast_client_cbs->location = NULL`, and then later assign a callback to it, but that is kind of a weird use case. 

Since this is a once-per-connection type of deal, I think it's OK to discover and subscribe to locations, regardless of the callback.